### ISSUE TITLE
[Feature] JSON support for generic, terraform and lint usage

### ISSUE THREAD

**Issue by doramatadora** [ysugimoto/falco#156]
**Describe the feature you'd like**
When `falco` is run without a subcommand, or with a subcommand other than `stat`, it should be possible to obtain JSON results for the run.

**Put example VCL after the feature has been implemented**
```
sub vcl_fetch {

  #Fastly fetch
  return(deliver)
}
```
Running `falco -json ./myVclFile.vcl` should result in:
```json
{
  "Infos": 0,
  "Warnings": 0,
  "Errors": 0,
  "LintErrors": {},
  "ParseErrors": {
    "/Users/dmilitaru/fsly/vscode-vcl/myVclFile.vcl": {
      "Token": {
        "Type": "RIGHT_PAREN",
        "Literal": ")",
        "Line": 4,
        "Position": 17,
        "Offset": 0,
        "File": "/Users/dmilitaru/fsly/vscode-vcl/myVclFile.vcl",
        "Snippet": false
      },
      "Message": "Missing semicolon"
    }
  },
  "Vcl": null
}
```

Running `falco -vv -json ./falco/examples/default03.vcl` should result in:
```json
{
  "Infos": 1,
  "Warnings": 0,
  "Errors": 0,
  "LintErrors": {
    "/Users/dmilitaru/fsly/vscode-vcl/falco/examples/default03.vcl": [
      {
        "Severity": "Info",
        "Token": {
          "Type": "REGEX",
          "Literal": "~",
          "Line": 34,
          "Position": 23,
          "Offset": 0,
          "File": "/Users/dmilitaru/fsly/vscode-vcl/falco/examples/default03.vcl",
          "Snippet": false
        },
        "Message": "regex captured variable may override older one",
        "Reference": "",
        "Rule": "regex/matched-value-override"
      }
    ]
  },
  "ParseErrors": {},
  "Vcl": {
      // PARSED VCL AST 
   }
}
```

**Additional context**
`falco` is amazing! I would like to use it in a Language Server I'm building for Fastly VCL.

**Comment by ysugimoto** [comment]
Hi @doramatadora, thanks for the suggestion!

> I would like to use it in a Language Server I'm building for Fastly VCL.

I'm excited about your thought! So, what do you think that we make a new subcommand `diagnosis` for the language server?
Then you can use `falco diagnosis ./falco/examples/default03.vcl` for the specific way and you can get JSON as you wrote:

```json
{
  "Infos": 1,
  "Warnings": 0,
  "Errors": 0,
  "LintErrors": {
    "/Users/dmilitaru/fsly/vscode-vcl/falco/examples/default03.vcl": [
      {
        "Severity": "Info",
        "Token": {
          "Type": "REGEX",
          "Literal": "~",
          "Line": 34,
          "Position": 23,
          "Offset": 0,
          "File": "/Users/dmilitaru/fsly/vscode-vcl/falco/examples/default03.vcl",
          "Snippet": false
        },
        "Message": "regex captured variable may override older one",
        "Reference": "",
        "Rule": "regex/matched-value-override"
      }
    ]
  },
  "ParseErrors": {},
  "Vcl": {
      // PARSED VCL AST 
   }
}
```

I think the diagnosis JSON would be generic for the other use case.
Let me know your opinions, thanks 🙂

**Comment by doramatadora** [comment]
Hi @ysugimoto, I am proposing that the flag `-json` becomes top-level, like `-vv` and `-v`, for ergonomics, so you could run `falco -json -vv`, `falco lint -json -vv`, or `falco stat -json -vv`. What do you think?

I am also happy to refactor https://github.com/ysugimoto/falco/pull/157 to restrict the `-json` flag to a new `diagnosis` and the existing `stat` command 😸 

This is a fantastic tool and JSON output, whatever the incantation, will help us tremendously to deliver an LS. Thank you!

**Comment by doramatadora** [comment]
The Vcl AST would be `null` if there is a ParseError.

**Comment by ysugimoto** [comment]
@doramatadora 

OK, I almost agree with you.
However I have just one thing to discuss - when the user supplies `-json` option, we could ignore the verbose level such as `-v` or `-vv`.
This is because I think we should be better to add all lint results to the output JSON.

I also will take a look at your PR, thank you.

**Comment by doramatadora** [comment]
You are right, I don't think there is any cause to suppress verbosity for JSON output. I will amend my PR 🙌🏼

**Comment by doramatadora** [comment]
@ysugimoto I updated the PR to reflect that `-json` mode is always very verbose (ignores `-v` and `-vv` flags).

**Comment by doramatadora** [comment]
This is an example of a use case that this feature enables: 
https://github.com/ysugimoto/falco/assets/12828487/1dd9b0bb-6b5c-4412-8ef6-0eb17e4fd6b9

**Comment by ysugimoto** [comment]
I have just released a new version https://github.com/ysugimoto/falco/releases/tag/v0.23.0

I found some conflict in https://github.com/ysugimoto/falco/pull/158 but it relates https://github.com/ysugimoto/falco/pull/157, we can release patch version after resolved them.

**Comment by doramatadora** [comment]
🎉  Amazing, many thanks 🤩  I have resolved the merge conflicts in https://github.com/ysugimoto/falco/pull/158

**Comment by ysugimoto** [comment]
merged #158 and released `v0.23.1`, thanks for your contribution!
